imkar/analytical/analytical.py

```diff
@@ -0,0 +1,169 @@
+import numpy as np
+import pyfar as pf
+import numbers
+
+def rectangular(frequency_vector, phi_vector, width, length, height_to_length):
+    """
+    This function computes the scattering coefficient of periodic rectangular
+    profiles, assuming that the extent is infinite.
+
+    This function is based on Ducourneau's method described by Prof. Embrechts
+    [#]_.
+
+    Parameters
+    ----------
+    frequency_vector : array, double
+        Vector of frequencies the scattering coefficient is computed at.
+    phi_vector : array, double
+        Vector of incidences phi the scattering coefficient is computed at.
+        Values between 0°-90° allowed.
+    width : double
+        Width of the rectangulars in m.
+    length : double
+        Length of the regtangulars in m.
+    height_to_length : double
+        Height to length ratio of the rectangulars.
+
+    Returns
+    -------
+    s : FrequencyData
+        The analytical solution for the scattering coefficient of a 
+        rectangular profile at the given incidences and frequencies.    
+
+    References
+    ----------
+    .. [#]J. Embrechts and A. Billon, "Theoretical Determination of the Random-
+    Incidence Scattering Coefficients of Infinite Rigid Surfaces with a 
+    Periodic Rectangular Roughness Profile", Acta Acustica united with Acustica,
+    Bd. 97, Nr. 4, S. 607-617, Juli 2011, doi:10.3813/AAA.918441
+
+    Comment
+    -------
+    In this solution the parameter r in equation (9) is also limited to 
+    -N <= r <= N. The width of the well is described by the width parameter and
+    replaces the parameter 2A.
+
+    """
+    #check inputs    
+    if not isinstance(frequency_vector, np.ndarray) or\
+        not isinstance(frequency_vector.item(0), numbers.Real):
+        raise TypeError("frequency_vector has to be an array of real numbers")
+    if frequency_vector.ndim>1:
+        raise ValueError("frequency_vector has to be 1-dimensional")
+    if not isinstance(phi_vector, np.ndarray) or \
+        not isinstance(phi_vector.item(0), numbers.Real):
```

Comment:
```suggestion
            not isinstance(phi_vector.item(0), numbers.Real):
```
one more intent is nessesary, due to python syntax